### PR TITLE
buildDotnetModule: restore and build testProjectFile

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
Some fairly straightforward changes to automatically build tests when they're not referenced in the base project file. Also enables running them for `jackett` and `opentabletdriver`. 

Noticed `jackett` being out of date when getting it to run the unit tests, so i've bumped it.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by IvarWithoutBones** [comment]
Result of `nixpkgs-review pr 151468` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>alttpr-opentracker</li>
    <li>btcpayserver</li>
    <li>depotdownloader</li>
    <li>discordchatexporter-cli</li>
    <li>jackett</li>
    <li>nbxplorer</li>
    <li>netcoredbg</li>
    <li>opentabletdriver</li>
    <li>python-language-server</li>
    <li>ryujinx</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 151468` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>alttpr-opentracker</li>
    <li>btcpayserver</li>
    <li>depotdownloader</li>
    <li>discordchatexporter-cli</li>
    <li>jackett</li>
    <li>nbxplorer</li>
    <li>netcoredbg</li>
    <li>opentabletdriver</li>
    <li>python-language-server</li>
    <li>ryujinx</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
I can only review the changes on OpenTabletDriver, would be interesting that someone can reviews the `buildDotnetModule` changes.

CC @prusnak.

**Comment by prusnak** [comment]
Please fix build failure of `jackett` on x86_64-darwin: https://github.com/NixOS/nixpkgs/pull/151468/checks?check_run_id=4588836733

The `jackett` also fail on aarch64-darwin, but that is because we don't have mono there, let's track that in a separate issue: https://github.com/NixOS/nixpkgs/issues/151720

Maybe we can just set `doCheck = !(stdenv.isDarwin && stdenv.isAarch64); # mono is not available on aarch64-darwin` ?

**Comment by IvarWithoutBones** [comment]
> Please fix build failure of jackett on x86_64-darwin: https://github.com/NixOS/nixpkgs/pull/151468/checks?check_run_id=4588836733 (`ln: failed to create symbolic link '/nix/store/ibqqs49yif3jjzs67smciba8fjqmwrwy-jackett-0.20.184/bin/Jackett': File exists`)

Huh, i didn't know executable name differentiates on linux and darwin. It should now always contain both the legacy and regular executable name on both platforms. Waiting on ofborg to build it, as I don't have a darwin machine.

> Maybe we can just set doCheck = !(stdenv.isDarwin && stdenv.isAarch64); # mono is not available on aarch64-darwin ?

Done :+1:

**Comment by prusnak** [comment]
@GrahamcOfBorg jackett opentabletdriver

**Comment by prusnak** [comment]
@GrahamcOfBorg build jackett opentabletdriver

File: pkgs/servers/jackett/default.nix [hunk 1]
```diff
@@ -1,32 +1,43 @@
...
+  checkInputs = [ mono ];
+  testProjectFile = "src/Jackett.Test/Jackett.Test.csproj";
+
+  postFixup = if (!stdenv.isDarwin) then ''
     # Legacy
     ln -s $out/bin/jackett $out/bin/Jackett
+  '' else ''
+    # For compatibility
+    ln -s $out/bin/Jackett $out/bin/jackett
   '';
```
Comment (prusnak) [inline pkgs/servers/jackett/default.nix:41]: ISTM that darwin build produces both binaries? https://github.com/NixOS/nixpkgs/pull/151468/checks?check_run_id=4606973723

I think we can drop the postFixup section completely (and expect user to just use `jackett` and not `Jackett`) or replace it with the following to allow both on all systems:

```suggestion
  postFixup =''
    # For compatibility
    ln -s $out/bin/jackett $out/bin/Jackett || :
    ln -s $out/bin/Jackett $out/bin/jackett || :
  '';
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/servers/jackett/default.nix (hunk 1 above)
Comment (IvarWithoutBones) [reply pkgs/servers/jackett/default.nix:41]: Sounds good, ill keep the alias for now as im not sure if anything depends on it.

**Comment by prusnak** [comment]
@GrahamcOfBorg build jackett opentabletdriver

**Comment by prusnak** [comment]
@GrahamcOfBorg build btcpayserver nbxplorer

**Comment by prusnak** [comment]
Thanks!

**Pull request merged**